src/should-delete-fork.js

```diff
@@ -18,131 +16,84 @@ const branchIsUseful = (github, repo, branch, parentBranches, callback) => {
 	const branchesToCheck = parentBranches.filter((b) => {
 		return [branch.name, 'master'].includes(b.name);
 	});
-	async.some(
-		branchesToCheck,
-		(candidate, someCallback) => {
-			github.repos.compareCommits(
-				{
-					owner: repo.parent.owner.login,
-					repo: repo.parent.name,
-					base: branch.commit.sha,
-					head: candidate.name
-				},
-				(error, diff) => {
-					if (error) {
-						// If diff can't be found, means our commit is not on this
-						// parent branch candidate
-						if (error.status === 404) {
-							return someCallback(null, false);
-						}
-
-						return someCallback(error);
-					}
-
-					// If parent is not behind us, this parent candidate
-					// branch contains our branch
-					someCallback(null, !diff.behind_by);
-				}
-			);
-		},
-		(error, parentContainsOurCommit) => {
-			if (error) {
-				callback(error);
-			} else {
-				callback(null, !parentContainsOurCommit);
+	for (const candidate of branchesToCheck) {
+		try {
+			// eslint-disable-next-line no-await-in-loop
+			const {data: diff} = await github.repos.compareCommits({
```

Comment:
The removal of all the custom code for GitHub processing is great. As a side effect, we now have `.data` field on the right again, but that is fine.

How come the tests are not affected? The fakes should be returning the values without the `data` field, how come they still pass? If you don’t know, I can look into them.

Reply:
I think the tests are around the mocks, which are being called just like before. They weren’t affected by what was in GitHub-factory

Reply:
`github.repos.compareCommits` above should be mocked and unchanged, correct. But the `const { data: diff }` above is new and should not work on the mocked response from `compareCommits`. I’ll take a look at it.

Reply:
I have looked more into it and my guess was incorrect. Our GitHub object always returned the object with the data field, and so did the tests. This PR changes the unwrapping to happen instead of in `nd` function in github-factory to manually unwrap where we use them. You’re right in that this has no effect, but now I also understand why!